src/swell/deployment/prepare_config_and_suite/prepare_config_and_suite.py

```diff
@@ -0,0 +1,601 @@
+                        model_dict[key].update(platform_defaults[key])
+
+        # Look for defer_to_code in the model_ind dictionary
+        # --------------------------------------------------
+        for key, val in self.question_dictionary_model_ind.items():
+            if val['default_value'] == 'defer_to_code':
+
+                if key == 'experiment_id':
+                    val['default_value'] = f'swell-{self.suite}'
+
+                if key == 'model_components':
+                    val['default_value'] = self.possible_model_components
+                    val['options'] = self.possible_model_components
+
+    # ----------------------------------------------------------------------------------------------
+
+    def override_with_external(self):
+
+        # Create and override dictionary
+        override_dict = {}
+
+        # Always start the override with the a suite test file
+        test_file = os.path.join(get_swell_path(), 'test', 'suite_tests',
+                                 self.suite + '-tier1.yaml')
+        if os.path.exists(test_file):
+            with open(test_file, 'r') as ymlfile:
+                override_dict = yaml.safe_load(ymlfile)
+
+        # Now append with any user provided override
+        if self.override is not None:
+
+            if isinstance(self.override, dict):
+                override_dict.update(self.override)
+            elif isinstance(self.override, str):
+                with open(self.override, 'r') as ymlfile:
+                    override_dict.update(yaml.safe_load(ymlfile))
+            else:
+                self.logger.abort(f'Override must be a dictionary or a path to a yaml file.')
+
+        # In this case the user is sending in a dictionary that looks like the experiment dictionary
+        # that they will ultimately be looking at. This means the dictionary does not contain
+        # default_value or options and the override cannot be performed.
+
+        # Iterate over the model_ind dictionary and override
+        # --------------------------------------------------
+        for key, val in self.question_dictionary_model_ind.items():
+            if key in override_dict.keys():
+                val['default_value'] = override_dict[key]
+
+        # Iterate over the model_dep dictionary and override
+        # --------------------------------------------------
+        if self.suite_needs_model_components:
+            for model, model_dict in self.question_dictionary_model_dep.items():
+                for key, val in model_dict.items():
+                    if model in override_dict['models'].keys():
+                        if key in override_dict['models'][model].keys():
+                            val['default_value'] = override_dict['models'][model][key]
+
+    # ----------------------------------------------------------------------------------------------
+
+    def ask_questions_and_configure_suite(self):
```

Comment:
As currently designed, the type annotation here should be:

```suggestion
    def ask_questions_and_configure_suite(self) -> Tuple(dict, dict):
```

However, the whole point of having the class is that we can just directly access the corresponding attributes in the calling code. I.e., We currently have:

```python
experiment_dict, comment_dict = prepper.ask_questions_and_configure_suite()
```

...when it is probably more idiomatic to have this method just modify the attributes (i.e., without a return statement) and then just grab the attributes directly. So, in the calling code:

```python
prepper.ask_questions_and_configure_suite()
experiment_dict = prepper.experiment_dict
comment_dict = prepper.comment_dict
```

(It's probably even _more_ idiomatic Python to just pass around the entire `prepper` object through various function calls. But that's a bigger change).

Reply:
I'm keeping the `Tuple` suggestion for now.